File: year2020/month10/day1008/ms_1008.cpp
#include <iostream>

using namespace std;

void half(int n);

int main() {
	int n;

	cin >> n;

	if (0 < n)
		half(n);
	
	return 0;
}

void half(int n) {
	if (n == 0)
		return;

	half(n / 2);
	cout << n << ' ';
}

File: year2020/month10/day1008/ms_1008.cs
using System;

namespace algorithm_74
{
    class Program
    {
        static void half(int n) 
        {
            if(n != 0) 
            {
                half(n/2);
                Console.Write($"{n}");
            }
        }

        static void Main(string[] args)
        {
            int n = Convert.ToInt32(Console.ReadLine());
            
            if(0 < n)
                half(n);
        }
    }
}
